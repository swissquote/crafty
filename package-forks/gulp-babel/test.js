const { Transform } = require("node:stream");
const { test } = require("node:test");
const { expect } = require("expect");
const path = require("path");
const Vinyl = require("vinyl");
const sourcemap = require("vinyl-sourcemap");
const babel = require(".");

test("should transpile with Babel", () => {
  return new Promise(done => {
    const stream = babel({
      plugins: ["@babel/transform-block-scoping"]
    });

    stream.on("data", file => {
      expect(/var foo/.test(file.contents.toString())).toBe(true);
      expect(file.relative).toBe("fixture.js");
    });

    stream.on("end", done);

    stream.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.jsx"),
        contents: Buffer.from("let foo;")
      })
    );

    stream.end();
  });
});

function initSourceMap() {
  return new Transform({
    objectMode: true,
    transform(file, enc, callback) {
      sourcemap.add(file, (sourcemapErr, updatedFile) => {
        if (sourcemapErr) {
          callback(sourcemapErr);
          return;
        }

        callback(null, updatedFile);
      });
    }
  });
}

function writeSourceMap() {
  return new Transform({
    objectMode: true,
    transform(file, enc, callback) {
      var self = this;

      sourcemap.write(file, (sourcemapErr, updatedFile, sourcemapFile) => {
        if (sourcemapErr) {
          callback(sourcemapErr);
          return;
        }

        self.push(updatedFile);
        if (sourcemapFile) {
          self.push(sourcemapFile);
        }

        callback();
      });
    }
  });
}

test("should generate source maps", () => {
  return new Promise(done => {
    const init = initSourceMap();
    const write = writeSourceMap();

    init
      .pipe(
        babel({
          plugins: ["@babel/transform-arrow-functions"]
        })
      )
      .pipe(write);

    write.on("data", file => {
      expect(file.sourceMap.sources).toEqual(["fixture.es2015"]);
      expect(file.sourceMap.file).toBe("fixture.js");
      const contents = file.contents.toString();
      expect(/function/.test(contents)).toBe(true);
      expect(/sourceMappingURL/.test(contents)).toBe(true);
      done();
    });

    init.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.es2015"),
        contents: Buffer.from("[].map(v => v + 1)"),
        sourceMap: ""
      })
    );

    init.end();
  });
});

test("should generate source maps for file in nested folder", () => {
  return new Promise(done => {
    const init = initSourceMap();
    const write = writeSourceMap();
    init
      .pipe(
        babel({
          plugins: ["@babel/transform-arrow-functions"]
        })
      )
      .pipe(write);

    write.on("data", file => {
      expect(file.sourceMap.sources).toEqual(["nested/fixture.es2015"]);
      expect(file.sourceMap.file).toBe("nested/fixture.js");
      const contents = file.contents.toString();
      expect(/function/.test(contents)).toBe(true);
      expect(/sourceMappingURL/.test(contents)).toBe(true);
      done();
    });

    init.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/nested/fixture.es2015"),
        contents: Buffer.from("[].map(v => v + 1)"),
        sourceMap: ""
      })
    );

    init.end();
  });
});

test("should pass the result of transform().metadata in file.babel", () => {
  return new Promise(done => {
    const stream = babel({
      plugins: [
        {
          post(file) {
            file.metadata.test = "metadata";
          }
        }
      ]
    });

    stream.on("data", file => {
      expect(file.babel).toEqual({ test: "metadata" });
    });

    stream.on("end", done);

    stream.write(
      new Vinyl({
        cwd: __dirname,
        base: path.join(__dirname, "fixture"),
        path: path.join(__dirname, "fixture/fixture.js"),
        contents: Buffer.from("class MyClass {};")
      })
    );

    stream.end();
  });
});

test("should not rename ignored files", () => {
  return new Promise(done => {
    const stream = babel({
      ignore: [/fixture/]
    });

    const inputFile = {
      cwd: __dirname
    };

    inputFile.base = path.join(inputFile.cwd, "fixture");
    inputFile.basename = "fixture.jsx";
    inputFile.path = path.join(inputFile.base, inputFile.basename);
    inputFile.contents = Buffer.from(";");

    stream
      .on("data", file => {
        expect(file.relative).toBe(inputFile.basename);
      })
      .on("end", done)
      .end(new Vinyl(inputFile));
  });
});

test("should not rename files without an extension", () => {
  return new Promise(done => {
    const stream = babel();

    const inputFile = {
      cwd: __dirname
    };

    inputFile.base = path.join(inputFile.cwd, "bin");
    inputFile.basename = "app";
    inputFile.path = path.join(inputFile.base, inputFile.basename);
    inputFile.contents = Buffer.from(";");

    stream
      .on("data", file => {
        expect(file.relative).toBe(inputFile.basename);
      })
      .on("end", done)
      .end(new Vinyl(inputFile));
  });
});
