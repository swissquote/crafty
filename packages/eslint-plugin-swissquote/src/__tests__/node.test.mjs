import test from "node:test";
import { expect } from "expect";
import initSnapshot from "@onigoetz/ntr-expect-snapshot";

import { prepareESLint } from "../../test_utils.js";

initSnapshot(import.meta.url);

const lint = prepareESLint("recommended", "node");

const orderedLint = prepareESLint("recommended", "node", {
  rules: {
    "@swissquote/swissquote/import/order": [
      "error",
      {
        alphabetize: { order: "asc", caseInsensitive: true }
      }
    ]
  }
});

test("Doesn't warn on console.log", async t => {
  const result = await lint(`console.log("Yeah");\n`);

  expect(result.messages.length).toEqual(0);
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(0);
});

test("Works with ES6", async t => {
  const result = await orderedLint(
    `
import fs from "node:fs";

import timer from "node:timer";

import http from "node:http";

const something = [fs, timer, http];

something.push("a value");

console.log(something);
`
  );

  expect(result.messages.length).toEqual(1);
  expect(result.messages[0].message).toEqual(
    "`node:http` import should occur before import of `node:timer`"
  );
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(1);
});

test("Works with commonjs", async t => {
  const result = await lint(
    `
const fs = require("node:fs");
const timer = require("node:timer");
const http = require("node:http");

const something = [fs, timer, http];

something.push("a value");

console.log(something);
`
  );

  expect(result.messages.length).toEqual(0);
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(0);
});

test("Works with TypeScript", async t => {
  const result = await lint(
    `
import useSWROrig from "swr";

export type UnPromisify<T> = T extends Promise<infer U> ? U : T;

export function obviousString(): string {
  const myString: string = "a string";
  console.log(myString);
  return myString;
}

export function useSWR<T, Error = any>(
  url
): {
  data?: T;
  error?: Error;
  revalidate: () => Promise<boolean>;
  isValidating: boolean;
} {
  // Skip it on the server side.
  if (typeof window === undefined) {
    return {
      async revalidate() {
        return true;
      },
      isValidating: true
    };
  }

  return useSWROrig<T, Error>(
    url,
    async (u): Promise<T> => {
      const result = await fetch(u);

      if (result.status == 200) {
        return await result.json();
      }

      throw new Error(result.statusText);
    },
    {}
  );
}
`,
    "utils.ts"
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toEqual(0);
  expect(result.errorCount).toEqual(5);
});
