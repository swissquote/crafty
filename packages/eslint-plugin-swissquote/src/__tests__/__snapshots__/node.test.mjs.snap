// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works with TypeScript//0`] = `Array [
  Object {
    "column": 7,
    "endColumn": 20,
    "endLine": 20,
    "line": 20,
    "message": "Unexpected constant binary expression. Compares constantly with the right-hand side of the \`===\`.",
    "messageId": "constantBinaryOperand",
    "nodeType": "UnaryExpression",
    "ruleId": "no-constant-binary-expression",
    "severity": 2,
  },
  Object {
    "column": 25,
    "endColumn": 34,
    "endLine": 20,
    "line": 20,
    "message": "Invalid typeof comparison value.",
    "messageId": "invalidValue",
    "nodeType": "Identifier",
    "ruleId": "valid-typeof",
    "severity": 2,
    "suggestions": Array [
      Object {
        "data": Object {
          "type": "undefined",
        },
        "desc": "Use \`\\"undefined\\"\` instead of \`undefined\`.",
        "fix": Object {
          "range": Array [
            429,
            438,
          ],
          "text": "\\"undefined\\"",
        },
        "messageId": "suggestString",
      },
    ],
  },
  Object {
    "column": 10,
    "endColumn": 20,
    "endLine": 29,
    "line": 29,
    "message": "React Hook \\"useSWROrig\\" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?",
    "nodeType": "Identifier",
    "ruleId": "@swissquote/swissquote/react-hooks/rules-of-hooks",
    "severity": 2,
  },
  Object {
    "column": 25,
    "endColumn": 27,
    "endLine": 34,
    "line": 34,
    "message": "Expected '===' and instead saw '=='.",
    "messageId": "unexpected",
    "nodeType": "BinaryExpression",
    "ruleId": "eqeqeq",
    "severity": 2,
  },
  Object {
    "column": 16,
    "endColumn": 35,
    "endLine": 35,
    "line": 35,
    "message": "Redundant use of \`await\` on a return value.",
    "messageId": "redundantUseOfAwait",
    "nodeType": "Identifier",
    "ruleId": "no-return-await",
    "severity": 2,
    "suggestions": Array [
      Object {
        "desc": "Remove redundant \`await\`.",
        "fix": Object {
          "range": Array [
            707,
            713,
          ],
          "text": "",
        },
        "messageId": "removeAwait",
      },
    ],
  },
]`;
